bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js

```diff
@@ -0,0 +1,72 @@
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { showFeatureDataFlyout } from './FeatureDataFlyout';
+
+class FeatureDataPluginUIHandler extends StateHandler {
+    constructor (plugin, mapModule, config) {
+        super();
+        this.setState({
+            activeTab: null
+        });
+        this.mapModule = mapModule;
+        this.addStateListener(() => this.updateFlyout());
+    }
+
+    getFeatureDataLayers () {
+        return Oskari.getSandbox()
+            .findAllSelectedMapLayers()
+            .filter(layer => layer.isVisibleOnMap() && layer.hasFeatureData && layer.hasFeatureData());
+    }
+
+    getFeaturesByLayerId (layerId) {
+        const featuresMap = this.mapModule.getVectorFeatures(null, { layers: [layerId] });
+        return featuresMap && featuresMap[layerId] ? featuresMap[layerId].features : null;
+    }
+
+    setActiveTab (layerId) {
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : null;
+        this.updateState({
+            activeLayerId: layerId,
+            activeLayerFeatures: features
+        });
+    }
+
+    updateStateAfterMapEvent () {
+        const featureDataLayers = this.getFeatureDataLayers() || [];
+        const layerId = featureDataLayers?.find(layer => layer.getId() === this.state.activeLayerId) ? this.state.activeLayerId : featureDataLayers.map(layer => layer.getId())[0] || null;
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : [];
+        this.updateState({
+            activeLayerId: layerId,
+            layers: featureDataLayers,
+            activeLayerFeatures: features
+        });
+    }
```

Comment:
The one-liner is a bit hard to read, but this is the intention?
```suggestion
    updateStateAfterMapEvent () {
        const featureDataLayers = this.getFeatureDataLayers() || [];
        // make sure our active layer in state is still valid
        let currentLayer = featureDataLayers.find(layer => layer.getId() === this.state.activeLayerId);
        if (!currentLayer && featureDataLayers.length) {
            // layer removed/need to update active layer to a new one
            currentLayer = featureDataLayers[0];
        }
        const features = currentLayer ? this.getFeaturesByLayerId(currentLayer.getId()) : [];
        this.updateState({
            activeLayerId: currentLayer.getId(),
            layers: featureDataLayers,
            activeLayerFeatures: features
        });
    }
```

Reply:
Refactored the oneliner to a more human-readable form